pennylane_lightning/src/Bindings.cpp

```diff
@@ -18,45 +18,65 @@
 #include "pybind11/pybind11.h"
 #include "pybind11/stl.h"
 
+/// @cond DEV
+namespace {
 using Pennylane::StateVector;
 using std::complex;
 using std::string;
 using std::vector;
-
 namespace py = pybind11;
 
-template <class T = double>
-static StateVector<T> create(const py::array_t<complex<T>> *numpyArray) {
+/**
+ * @brief Create a `%StateVector` object from a 1D numpy complex data array.
+ *
+ * @tparam fp_t Precision data type
+ * @param numpyArray Numpy data array.
+ * @return StateVector<fp_t> `%StateVector` object.
+ */
+template <class fp_t = double>
```

Comment:
What is the relevance of `fp_t`?

Reply:
I had thought of using `fp_t` as floating point type template type, with a default value of double. Happy to rename if you think something more relevant makes sense.